consensus/state.go

```diff
@@ -669,9 +669,17 @@ func (cs *State) updateToState(state sm.State) {
 		// to be gathered for the first block.
 		// And alternative solution that relies on clocks:
 		// cs.StartTime = state.LastBlockTime.Add(timeoutCommit)
-		cs.StartTime = cs.config.Commit(cmttime.Now())
+		cs.StartTime = cs.config.NextStartTime(cmttime.Now())
 	} else {
-		cs.StartTime = cs.config.Commit(cs.CommitTime)
+		elapsed := cs.CommitTime.Sub(cs.StartTime)
+		nst := cs.config.TargetHeightDuration - elapsed
+		if nst < time.Millisecond*1 || elapsed < 0 {
+			nst = time.Millisecond * 1
+		}
+		cs.eventCollector.WritePoint("consensus", map[string]interface{}{
+			"elapsed_info": []interface{}{cs.Height, cs.Round, elapsed, nst},
+		})
+		cs.StartTime = cs.CommitTime.Add(nst)
```

Comment:
I'll abstract this out and unit test it, then we still need to run testground tests, along with an overally analysis of selecting ideal timings, but this is what is required to get consistent block times in the e2e test. (I'll aslo remove the event collector)